src/Bookings.js

```diff
@@ -1,21 +1,83 @@
-import React, { Component } from "react";
-import Search from "./Search.js";
-// import SearchResults from './SearchResults.js';
-// import FakeBookings from '../data/fakeBookings.json';
+import React, { Component } from 'react'
+import Search from './Search.js'
+import SearchResults from './SearchResults.js'
+import NewBooking from './NewBooking.js'
 
 export default class Bookings extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      FakeBookings: [],
+      error: null,
+      isLoading: true
+    }
+  }
+  componentDidMount () {
+    fetch('https://cyf-react.glitch.me/')
+      .then(res => {
+        if (res.status >= 200 && res.status < 300) {
+          return res
+        } else {
+          throw new Error('HTTP error')
+        }
+      })
+      .then(res => res.json())
+      .then(data => {
+        console.log(data)
+        this.setState({
+          isLoading: false,
+          FakeBookings: data
+        })
+      })
+      .catch(error => {
+        this.setState({
+          isLoading: false,
+          error: error
+        })
+      }),
+    1000
+  }
   search = searchVal => {
-    console.info("TO DO!", searchVal);
-  };
-
-  render() {
-    return (
-      <div className="App-content">
-        <div className="container">
-          <Search search={this.search} />
-          {/* <SearchResults results={FakeBookings} /> */}
+    if (searchVal !== '' || searchVal !== null) {
```

Comment:
Well done spotting the problem here. There may be a slightly improved approach. We generally say in React that it is better to calculate things when we need them rather than try to store everything in state. Here we only need to store the current value and the full list of results to be able to calculate the filtered list of results.

It might be easier to store just the current `searchVal` in state, then calculate the filtered results in the render method. Then if you want to show all the results again, you can just set the `searchVal` to be empty and then the filter will return all the results.

Reply:
I tried this with nested filter and map statement but when searchVal is ' ' or null filter return empty array means nothing to map where I want filter to return all the results ..... Can you explain a bit more your approach please

Reply:
I was thinking something like this: https://gist.github.com/40thieves/8f42634bdb49b9be5a31768a32645cc3

So yes you are correct that you need to handle empty strings or `null`, but my point was more about what you are storing in state.

In my approach, I'm storing **all** the results and the search string. Then we compute the currently filtered results in the `render` method. This is generally a simpler approach because we don't have to worry about keeping the state correct at all times. So for example if we added another search field (e.g. dates) then we'd have to make sure that we updated the state correctly to filter by name and by date. In my approach you would only need to update the component store the date that the user had entered  and update the `filterResults` method to also check that value in the filter.

This is why we generally say if it is possible to **compute** a value at render time that is the better approach.

Does that make more sense?